docs/src/test/java/docs/http/javadsl/server/HttpServerExampleDocTest.java

```diff
@@ -321,6 +322,35 @@ void discardEntityManuallyCloseConnections() {
       );
         //#discard-close-connections
       }
-    
-  
+
+  public static void gracefulTerminationExample() throws Exception {
+    //#graceful-termination
+    ActorSystem system = ActorSystem.create();
+    Materializer materializer = ActorMaterializer.create(system);
+
+
+    CompletionStage<ServerBinding> binding = Http.get(system).bindAndHandle(
+        Directives.complete("Hello world!").flow(system, materializer),
+        ConnectHttp.toHost("localhost", 8080), materializer);
+
+    ServerBinding serverBinding = binding.toCompletableFuture().get(3, TimeUnit.SECONDS);
+
+    // ...
+    // once ready to terminate the server, invoke terminate:
+    CompletionStage<HttpTerminated> onceAllConnectionsTerminated =
+        serverBinding.terminate(Duration.ofSeconds(3));
+
+    // once all connections are terminated,
+    HttpTerminated httpTerminated = onceAllConnectionsTerminated.toCompletableFuture().get();
```

Comment:
blocking `get()` (also without timeout) ?

Reply:
On purpose did not do callbacks since this is "where you start the server" where you often have a blocking read line or something anyway. But ok, can go back to callbacks

Reply:
this is not when starting the server, why would you terminate right after you have started it. Probably called from elsewhere, an actor, an incoming request, management...